soc/arm/st_stm32/stm32f7/soc.c

```diff
@@ -30,12 +30,12 @@ static int st_stm32f7_init(void)
 	/* Enable ART Flash cache accelerator */
 	LL_FLASH_EnableART();
 
-	SCB_EnableICache();
+	if (IS_ENABLED(CONFIG_ICACHE)) {
```

Comment:
Please mention in commit message that this has no impact on default configuration due to target selecting the right symbol ...

Reply:
Thanks for the suggestion. Done!